taichi/backends/cuda/cupti_toolkit.h

```diff
@@ -1,20 +1,29 @@
 #pragma once
 
-// #include "taichi/backends/cuda/cuda_profiler.h"
-#include "taichi/common/core.h"
-
-#include <string>
-#include <stdint.h>
+#include "taichi/program/kernel_profiler.h"
 
 TLANG_NAMESPACE_BEGIN
 
+// make sure these metrics can be captured in one pass (no kernal replay)
+enum CuptiMetricsDefault {
+  CUPTI_METRIC_KERNEL_ELAPSED_CLK_NUMS = 0,
+  CUPTI_METRIC_CORE_FREQUENCY_HZS = 1,
+  CUPTI_METRIC_GLOBAL_LOAD_BYTES = 2,
+  CUPTI_METRIC_GLOBAL_STORE_BYTES = 3,
+  CUPTI_METRIC_DEFAULT_TOTAL = 4
+};
+
+const std::vector<std::string> MetricListDeafult = {
```

Comment:
nit: `extern` this, then put the definition in the CPP file, e.g. https://stackoverflow.com/a/4391523/12003165 (Also can we just use an array of `const char*`s? E.g. `const char* kMetricListDeafult[]`?

Reply:
I'll remove this in the next PR, `MetricListDeafult` will be declared in Python scope [link](https://github.com/yolo2themoon/taichi/blob/to_merge/python/taichi/profiler/kernelmetrics.py#L125).

Reply:
Putting constant definition in header is dangerous. Let's just get rid of it in one PR :-)